src/functional/ffront/foast_to_itir.py

```diff
@@ -170,28 +218,36 @@ class InsideReductionLowering(FieldOperatorLowering):
     lambda_params: dict[str, itir.Expr] = field(default_factory=lambda: {})
     __counter: itertools.count = field(default_factory=lambda: itertools.count())
 
-    def visit_Name(self, node: foast.Name, **kwargs) -> itir.SymRef:
+    def visit_Name(self, node: foast.Name, *, to_value: bool = False, **kwargs) -> itir.SymRef:
         uid = f"{node.id}__{self._sequential_id()}"
         self.lambda_params[uid] = super().visit_Name(node, **kwargs)
         return im.ref(uid)
 
-    def visit_BinOp(self, node: foast.BinOp, **kwargs) -> itir.FunCall:
+    def visit_BinOp(self, node: foast.BinOp, *, to_value: bool = False, **kwargs) -> itir.FunCall:
         return im.call_(node.op.value)(
             self.visit(node.left, **kwargs), self.visit(node.right, **kwargs)
         )
 
-    def visit_Compare(self, node: foast.Compare, **kwargs) -> itir.FunCall:
+    def visit_Compare(
+        self, node: foast.Compare, *, to_value: bool = False, **kwargs
+    ) -> itir.FunCall:
         return im.call_(node.op.value)(
             self.visit(node.left, **kwargs), self.visit(node.right, **kwargs)
         )
 
-    def visit_UnaryOp(self, node: foast.UnaryOp, **kwargs) -> itir.FunCall:
-        zero_arg = [itir.IntLiteral(value=0)] if node.op is not foast.UnaryOperator.NOT else []
+    def visit_UnaryOp(
+        self, node: foast.UnaryOp, *, to_value: bool = False, **kwargs
+    ) -> itir.FunCall:
+        zero_arg = (
+            [itir.NumberLiteral(value="0", type="int")]
+            if node.op is not foast.UnaryOperator.NOT
+            else []
+        )
         return im.call_(node.op.value)(*[*zero_arg, self.visit(node.operand, **kwargs)])
 
-    def _visit_shift(self, node: foast.Call, **kwargs) -> itir.SymRef:  # type: ignore[override]
+    def _visit_shift(self, node: foast.Call, *, to_value: bool = False, **kwargs) -> itir.SymRef:  # type: ignore[override]
         uid = f"{node.func.id}__{self._sequential_id()}"
-        self.lambda_params[uid] = FieldOperatorLowering.apply(node, **kwargs)
+        self.lambda_params[uid] = FieldOperatorLowering.apply(node)
```

Comment:
Not really, `FieldOperatorLowering.apply` does not take any keyword arguments. This bug just never caused any problems because ``kwargs`` is usually empty.

Reply:
Then I would add an assert that they are actually empty.

Reply:
I wouldn't, at least not without a compelling reason.

Reply:
I should also mention here that a "vanilla" `FieldOperatorLowering` is applied here on purpose. Even if there was a way to pass on any keyword arguments here, it should not be done.

In other words kwargs should be allowed to reach this method but should not be passed on.